TowerNinja/Assets/Scripts/TowerBehaviourScript.cs

```diff
@@ -114,6 +116,21 @@ private void UpdateTowerAppearance()
         if (hpFloored < 3) _levels[2].SetActive(false);
         if (hpFloored < 2) _levels[1].SetActive(false);
         if (hpFloored < 1) _levels[0].SetActive(false);
+
+        // adjust collider shape
+        if (hpFloored >= 9) UpdateTowerColliderShape(9, -3);
+        else UpdateTowerColliderShape(hpFloored, -3.0f - 0.5f * (9 - hpFloored));
```

Comment:
What does `-3.0f - 0.5f * (9 - hpFloored)` do/mean

Reply:
This is hard-coded based on our current tower setup. Initially the tower/collider has a size of (3, 9) and a position offset of (-1, -3). As the tower becomes smaller, the size moves from (3, 9) -> (3, 8) -> (3, 7) ->..... -> (3, 0). The offset moves from (-1, -3.5) -> (-1, -4) -> (-1, -4.5) -> .... -> (-1, -7.5)